src/main/telemetry/ibus_shared.c

```diff
@@ -57,120 +79,460 @@ typedef enum {
+static uint8_t transmitIbusPacket()
 {
-    uint16_t checksum = calculateChecksum(ibusPacket, payloadLength + IBUS_CHECKSUM_SIZE);
+    uint8_t frameLength = sendBuffer[0];
+    if (frameLength == INVALID_IBUS_ADDRESS) {
+        return 0;
+    }
+    uint8_t payloadLength = frameLength - IBUS_CHECKSUM_SIZE;
+    uint16_t checksum = calculateChecksum(sendBuffer);
     for (size_t i = 0; i < payloadLength; i++) {
-        serialWrite(ibusSerialPort, ibusPacket[i]);
+        serialWrite(ibusSerialPort, sendBuffer[i]);
     }
     serialWrite(ibusSerialPort, checksum & 0xFF);
     serialWrite(ibusSerialPort, checksum >> 8);
-    return payloadLength + IBUS_CHECKSUM_SIZE;
+    return frameLength;
 }
 
-static uint8_t sendIbusDiscoverSensorReply(ibusAddress_t address)
+static void setIbusDiscoverSensorReply(ibusAddress_t address)
 {
-    uint8_t sendBuffer[] = { 0x04, IBUS_COMMAND_DISCOVER_SENSOR | address};
-    return transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+    sendBuffer[0] = IBUS_HEADER_FOOTER_SIZE;
+    sendBuffer[1] = IBUS_COMMAND_DISCOVER_SENSOR | address;
 }
 
-static uint8_t sendIbusSensorType(ibusAddress_t address)
+static void setIbusSensorType(ibusAddress_t address)
 {
-    uint8_t sendBuffer[] = {0x06,
-                            IBUS_COMMAND_SENSOR_TYPE | address,
-                            sensorAddressTypeLookup[address - ibusBaseAddress],
-                            0x02
-                           };
-    return transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+    uint8_t sensorID = getSensorID(address);
+    uint8_t sensorLength = getSensorLength(sensorID);
+    sendBuffer[0] = IBUS_HEADER_FOOTER_SIZE + 2;
+    sendBuffer[1] = IBUS_COMMAND_SENSOR_TYPE | address;
+    sendBuffer[2] = sensorID;
+    sendBuffer[3] = sensorLength;
 }
 
-static uint8_t sendIbusMeasurement(ibusAddress_t address, uint16_t measurement)
-{
-    uint8_t sendBuffer[] = { 0x06, IBUS_COMMAND_MEASUREMENT | address, measurement & 0xFF, measurement >> 8};
-    return transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+static uint16_t getVoltage(){
+    uint16_t voltage = getBatteryVoltage() *10;
+    if (telemetryConfig()->report_cell_voltage) {
+        voltage /= getBatteryCellCount();
+    }
+    return voltage;
 }
 
-static bool isCommand(ibusCommand_e expected, const uint8_t *ibusPacket)
-{
-    return (ibusPacket[1] & 0xF0) == expected;
+static uint16_t getTemperature() {
+    uint16_t temperature = gyroGetTemperature() * 10;
+#if defined(BARO)
+    if (sensors(SENSOR_BARO)) {
+        temperature = (uint16_t) ((baro.baroTemperature + 50) / 10);
+    }
+#endif
+    return temperature + IBUS_TEMPERATURE_OFFSET;
 }
 
-static ibusAddress_t getAddress(const uint8_t *ibusPacket)
-{
-    return (ibusPacket[1] & 0x0F);
+
+static uint16_t getFuel(){
+    uint16_t fuel = 0;
+    if (batteryConfig()->batteryCapacity > 0) {
+        //float value = constrain(batteryConfig()->batteryCapacity - getMAhDrawn(), 0, batteryConfig()->batteryCapacity);
```

Comment:
What is this commented out code for? Commented out code should either be annotated with why it's in there, or removed, to avoid confusion.

Reply:
OK, removed.